CSE/Logic/Database/Reader.cs

```diff
@@ -28,16 +29,41 @@ public void CloseConnection()
 			con.Close();
 		}
 
-		public IEnumerable<string> ReadData(string query)
+		public IEnumerable<string> ReadProductData()
 		{
-			List<string> list = new List<string>();
-			SqlCommand cmd = new SqlCommand(query, con);
+			List<string> productList = new List<string>();
+			SqlCommand cmd = new SqlCommand("SELECT DISTINCT name FROM product", con);
 			reader = cmd.ExecuteReader();
 			while (reader.Read())
 			{
-				list.Add(reader.GetString(0).ToString());
+				productList.Add(reader.GetString(0));
 			}
-			return list;
+			return productList;
+		}
+
+		public List<Tuple<decimal, DateTime>> ReadHistoryData(int storeName, string productName)
+		{
+			var history = new List<Tuple<decimal, DateTime>>();
```

Comment:
It should be:
var history = new List<Dictionary<DateTime, decimal>>();
It's because i use this type in GraphOperations

Reply:
Cannot use dictionary because it contains a key which means each key has to be unique whereas in some instances price or/and date might contain duplicate values.

Reply:
Okey, i'll change mine. But change the order in Tuple if you can.